stack/jenkins-stack-os-matrix.rst

```diff
@@ -20,18 +20,25 @@ Running a stack-os-matrix job
 
      For example, if you enter ``tickets/DM-2 tickets/DM-1``, the build system will attempt to check out the ``tickets/DM-2`` branch in Stack packages.
      If a package does not have a ``tickets/DM-2`` branch, it will attempt to check out the ``tickets/DM-1`` branch.
+
      If a package has neither branch, it falls back to checking out the ``main`` branch or the branch configured in ``repos.yaml`` in the case of forked third-party packages.
      Because of those third-party packages, you *never* want to specify ``main`` explicitly in this field.
+     To check that a ``main``-only build passes, leave the refs box entirely blank.
 
    - **Set the list of EUPS packages** to build.
-     Use the default (``lsst_distrib``) to build and test your changes with a full Stack.
-     To improve build times you can instead specify the name of the package you are actively developing and check the **Skip Demo** box.
+     Use the default (``lsst_distrib lsst_ci``) to build and test your changes with a full Stack.
+     To improve build times you can instead specify the name of the package you are actively developing.
+     Before you merge a ticket branch to ``main``, **you must** run a stack-os-matrix Job with at least the default packages so that the full Stack is built and tested with your changes.
 
-   - **Check the Skip Demo** option only if you are testing your package alone (in conjunction with specifying your package in the prior text box).
+     There are also several packages that can be appended to the default to do more thorough testing at the cost of much longer build times:
 
-     Before you merge a ticket branch to ``main``, **you must** run a stack-os-matrix Job without **Skip Demo** enabled so that the full Stack is built and tested with your changes.
+     - ``ci_cpp`` exercises the Calibration Products Pipeline in ways that are too computationally expensive for unit tests in ``cp_pipe``.
+     - ``ci_hsc`` exercises most of the Data Release Production pipelines, including single-frame processing and coaddition, on HSC engineering data.
```

Comment:
Late comment -- do we want to give guidance for when someone has to do this? Do science pipelines want at least one of ci_hsc or ci_imsim to be run for every single change to a pipelines package?

Reply:
My intent in describing what the packages do was that that would _be_ the guidance.

I would oppose a rule that broad on the grounds that it makes things unnecessarily hard to merge. Personally, I only run the big packages when I have a change that I'd reasonably expect to affect the pipelines (e.g., changes to `Exposure`, but not `ap_verify`).

In any case, we have never had such a policy before, so I think adopting it would require an RFC.